### PR TITLE
[Docs] Smallish overhaul of the docs

### PR DESCRIPTION
**Pull request by drashna**
## Description

* Fix up Common Features list
   * Convert AVR Pin functions to GPIO commands
   * Fix EEPROM Example to use `keyboard_post_init_user`
* Add additional supported commands to Flashing doc
* Add additional supported commands to newb flashing doc
* Rearrange Macro Feature page, and add additional supported commands
* Update Process_record list
* Rename Best Practices link in sidebar to `Git Best Practices`
* Expand FAQ Build guide's ZADIG section
* Add Deprecation info to keymap doc

Basically, this is a collection of small changes that have been bugging me for a while. 

## Types of Changes
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).

### CODE REVIEW

File: docs/feature_macros.md [hunk 1]
```diff
@@ -146,9 +146,59 @@ send_string(my_str);
...
+### `clear_mods();`
+
+This will clear all mods currently pressed.
+
+### `clear_keyboard_but_mods();`
+
+This will clear all keys besides the mods currently pressed.
+
+
+##  **(DEPRECIATED)** The Old Way: `MACRO()` & `action_get_macro`
```
Comment (fauxpark) [inline docs/feature_macros.md]:
```suggestion
##  **(DEPRECATED)** The Old Way: `MACRO()` & `action_get_macro`
```

[This page](https://www.dailywritingtips.com/depreciate-vs-deprecate/) seems to conclude that "deprecated" would be more correct here.

File: docs/feature_macros.md [hunk 2]
```diff
@@ -146,9 +146,59 @@ send_string(my_str);
...
+
+!> This is inherited from TMK, and hasn't been updated - it's recommend that you use `SEND_STRING` and `process_record_user` instead.
```
Comment (fauxpark) [inline docs/feature_macros.md]:
```suggestion
!> This is inherited from TMK, and hasn't been updated - it's recommended that you use `SEND_STRING` and `process_record_user` instead.
```

File: docs/flashing.md [hunk 3]
```diff
@@ -49,6 +49,15 @@ To generate this bootloader, use the `bootloader` target, eg `make planck/rev4:d
 
 To generate a production-ready .hex file (containing the application and the bootloader), use the `production` target, eg `make planck/rev4:default:production`.
 
+### DFU commands
+
+There are a number of DFU commands that you can use to flash firmware to a DFU device
```
Comment (fauxpark) [inline docs/flashing.md]:
```suggestion
There are a number of DFU commands that you can use to flash firmware to a DFU device:
```

File: docs/flashing.md [hunk 4]
```diff
@@ -131,3 +141,12 @@ Flashing sequence:
     * You will receive a warning about the DFU signature; Just ignore it
 4. Reset the device into application mode (may be done automatically)
     * If you are building from command line (e.g. `make planck/rev6:default:dfu-util`), make sure that `:leave` is passed to the `DFU_ARGS` variable inside your `rules.mk` (e.g. `DFU_ARGS = -d 0483:df11 -a 0 -s 0x08000000:leave`) so that your device resets after flashing
+
+### STM32 Commands
+
+There are a number of DFU commands that you can use to flash firmware to a STM32 device
```
Comment (fauxpark) [inline docs/flashing.md]:
```suggestion
There are a number of DFU commands that you can use to flash firmware to a STM32 device:
```

File: docs/keymap.md [hunk 5]
```diff
@@ -161,7 +161,11 @@ Some interesting things to note:
 * We have used our `_______` definition to turn `KC_TRNS` into `_______`. This makes it easier to spot the keys that have changed on this layer.
 * While in this layer if you press one of the `_______` keys it will activate the key in the next lowest active layer.
 
-### Custom Functions
+### **(DEPRECIATED)** Custom Functions
```
Comment (fauxpark) [inline docs/keymap.md]:
```suggestion
### **(DEPRECATED)** Custom Functions
```

And here too ;)

File: docs/newbs_flashing.md [hunk 6]
```diff
@@ -131,6 +131,16 @@ If you have any issues with this, you may need to this:
 
     sudo make <my_keyboard>:<my_keymap>:dfu
 
+#### DFU commands
+
+There are a number of DFU commands that you can use to flash firmware to a DFU device
```
Comment (fauxpark) [inline docs/newbs_flashing.md]:
```suggestion
There are a number of DFU commands that you can use to flash firmware to a DFU device:
```

File: docs/newbs_flashing.md [hunk 7]
```diff
@@ -199,6 +209,14 @@ If you have any issues with this, you may need to this:
 
     sudo make <my_keyboard>:<my_keymap>:avrdude
 
+
+Additionally, if you want to flash multiple boards, use the following command
```
Comment (fauxpark) [inline docs/newbs_flashing.md]:
```suggestion
Additionally, if you want to flash multiple boards, use the following command:
```

File: docs/faq_build.md [hunk 8]
```diff
@@ -47,7 +47,10 @@ If you're using Windows to flash your keyboard, and you are running into issues,
 
 Re-running the installation script for MSYS2 may help (eg run `./util/qmk_install.sh` from MSYS2/WSL) or reinstalling the QMK Toolbox may fix the issue.
 
-If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUS(libusb-1.0)` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.
+If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUS(libusb-1.0)` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.  If that doesn't work, try all of the options, until one works.
```
Comment (noroadsleft) [inline docs/faq_build.md]: Current Zadig (v2.4, Build 721) gives the following options for me:

- `WinUSB (v6.1.7600.16385)`
- `libusb-win32 (v1.2.6.0)`
- `libusbK (v3.0.7.0)`
- `USB Serial (CDC)`

```suggestion
If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUSB` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.  If that doesn't work, try all of the options, until one works. 
```

Used `WinUSB` without a specified version number in case a new version of Zadig Utility uses a different version of the driver.

File: docs/feature_macros.md [hunk 9]
```diff
@@ -146,9 +146,59 @@ send_string(my_str);
 SEND_STRING(".."SS_TAP(X_END));
 ```
 
-## The Old Way: `MACRO()` & `action_get_macro`
 
-?> This is inherited from TMK, and hasn't been updated - it's recommend that you use `SEND_STRING` and `process_record_user` instead.
+## Advanced Macro Functions
+
+There are some functions you may find useful in macro-writing. Keep in mind that while you can write some fairly advanced code within a macro if your functionality gets too complex you may want to define a custom keycode instead. Macros are meant to be simple.
```
Comment (noroadsleft) [inline docs/feature_macros.md]:
```suggestion
There are some functions you may find useful in macro-writing. Keep in mind that while you can write some fairly advanced code within a macro, if your functionality gets too complex you may want to define a custom keycode instead. Macros are meant to be simple.
```

File: docs/keymap.md [hunk 10]
```diff
@@ -161,7 +161,11 @@ Some interesting things to note:
...
+
+<!-- Maybe we should remove this section, since it's very depreciated -->
+
+!> Functions are depreciated in favor of [custom macros](feature_macros.md) or have been rolled into keycodes.
```
Comment (noroadsleft) [inline docs/keymap.md]:
```suggestion
!> These functions are deprecated in favor of [custom macros](feature_macros.md) or have been rolled into keycodes.  
```

File: docs/custom_quantum_functions.md [hunk 11]
```diff
@@ -196,9 +196,11 @@ void keyboard_pre_init_user(void) {
   // Call the keyboard pre init code.
 
   // Set our LED pins as output
-  DDRB |= (1<<1);
-  DDRB |= (1<<2);
-  DDRB |= (1<<3);
+  setPinOutput(B0);
```
[the remark this answers is not in the document]
Comment (drashna) [reply docs/custom_quantum_functions.md]: Updated text.

File: docs/custom_quantum_functions.md [hunk 12]
```diff
@@ -374,7 +373,7 @@ void matrix_init_user(void) {
   }
 }
 ```
-The above function will use the EEPROM config immediately after reading it, to set the default layer's RGB color. The "raw" value of it is converted in a usable structure based on the "union" that you created above. 
+The above function will use the EEPROM config immediately after reading it, to set the default layer's RGB color. The "raw" value of it is git converted in a usable structure based on the "union" that you created above.
```
Comment (noroadsleft) [inline docs/custom_quantum_functions.md]: Uhhh... what?

```suggestion
The above function will use the EEPROM config immediately after reading it, to set the default layer's RGB color. The "raw" value of it is converted in a usable structure based on the "union" that you created above. 
```

???

Comment (drashna) [reply docs/custom_quantum_functions.md]: Because of the stupid gpg prompt .....

File: docs/custom_quantum_functions.md [hunk 13]
```diff
@@ -116,29 +116,29 @@ Use the `IS_LED_ON(usb_led, led_name)` and `IS_LED_OFF(usb_led, led_name)` macro
 ```c
 void led_set_user(uint8_t usb_led) {
     if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
-        PORTB |= (1<<0);
+        writePinLow(B0);
```
Comment (fauxpark) [inline docs/custom_quantum_functions.md]: `|=` sets the bit, so these should all be flipped, no?

Comment (drashna) [reply docs/custom_quantum_functions.md]: Yeah, it depends on how the LED is wired.   And if the pin is sit to input/output.

File: docs/faq_build.md [hunk 14]
```diff
@@ -47,7 +47,10 @@ If you're using Windows to flash your keyboard, and you are running into issues,
 
 Re-running the installation script for MSYS2 may help (eg run `./util/qmk_install.sh` from MSYS2/WSL) or reinstalling the QMK Toolbox may fix the issue.
 
-If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUS(libusb-1.0)` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.
+If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUSB` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.  If that doesn't work, try all of the options, until one works.
```
Comment (mechmerlin) [inline docs/faq_build.md]: The libusb option actually works better.

Comment (drashna) [reply docs/faq_build.md]: It doesn't for me, actually.  In fact, the WinUSB was the only one that really works for me. 

Hence why I mention using all of them.  But I can change this.

Comment (drashna) [reply docs/faq_build.md]:
```suggestion
If that doesn't work, then you may need to grab the [Zadig Utility](https://zadig.akeo.ie/). Download this, find the device in question, and select the `WinUSB` option, and hit "Reinstall driver". Once you've done that, try flashing your board, again.  If that doesn't work, try all of the options, until one works. 

?> There isn't a best option for which driver should be used here.  Some options work better on some systems than others.  libUSB and WinUSB seem to be the best options here.
```

File: docs/keymap.md [hunk 15]
```diff
@@ -161,7 +161,11 @@ Some interesting things to note:
 * We have used our `_______` definition to turn `KC_TRNS` into `_______`. This makes it easier to spot the keys that have changed on this layer.
 * While in this layer if you press one of the `_______` keys it will activate the key in the next lowest active layer.
 
-### Custom Functions
+### **(DEPRECATED)** Custom Functions
+
+<!-- Maybe we should remove this section, since it's very depreciated -->
```
Comment (nooges) [inline docs/keymap.md]: s/depreciated/deprecated

File: docs/keymap.md [hunk 16]
```diff
@@ -161,7 +161,11 @@ Some interesting things to note:
 * We have used our `_______` definition to turn `KC_TRNS` into `_______`. This makes it easier to spot the keys that have changed on this layer.
 * While in this layer if you press one of the `_______` keys it will activate the key in the next lowest active layer.
 
-### Custom Functions
+### **(DEPRECATED)** Custom Functions
+
+<!-- Maybe we should remove this section, since it's very deprecated -->
```
Comment (ezuk) [inline docs/keymap.md]: why not go ahead and do this, remove it?

Comment (drashna) [reply docs/keymap.md]: Well, it's used in some keymaps, so I wasn't sure it should be completely removed, yet. 

But yeah, I do want to completely remove this.
